apps/dashboard/src/hooks/use-onboarding-steps.ts

```diff
@@ -0,0 +1,136 @@
+  return providerType === ChannelTypeEnum.IN_APP
+    ? 'Add an Inbox to your app'
+    : `Connect your provider to send ${providerType} notifications with Novu.`;
+}
+
+function isActiveIntegration(integration: IIntegration, providerType: ChannelTypeEnum): boolean {
+  const isMatchingChannel = integration.channel === providerType;
+  const isNotNovuProvider = !integration.providerId.startsWith('novu-');
+  const isConnected = providerType === ChannelTypeEnum.IN_APP ? !!integration.connected : true;
+
+  return isMatchingChannel && isNotNovuProvider && isConnected;
+}
+
+export function useOnboardingSteps(): OnboardingStepsResult {
+  const { data: workflows } = useWorkflows();
+  const { organization } = useOrganization();
+  const { integrations } = useIntegrations();
+  const telemetry = useTelemetry();
+  const previousStepsRef = useRef<Step[]>([]);
+
+  const hasInvitedTeamMember = useMemo(() => {
+    return (organization?.membersCount ?? 0) > 1;
+  }, [organization?.membersCount]);
+
+  const providerType = useMemo(() => {
+    const metadata = organization?.publicMetadata as OrganizationMetadata;
+    const useCases = metadata?.useCases ?? DEFAULT_USE_CASES;
+
+    return PROVIDER_TYPE_PRIORITIES.find((type) => useCases.includes(type)) ?? useCases[0];
+  }, [organization?.publicMetadata]);
+
+  const steps = useMemo(
+    (): Step[] => [
+      {
+        id: StepIdEnum.ACCOUNT_CREATION,
+        title: 'Account creation',
+        description: "We know it's not always easy — take a moment to celebrate!",
+        status: 'completed',
+      },
+      {
+        id: StepIdEnum.CREATE_A_WORKFLOW,
+        title: 'Create a workflow',
+        description: 'Workflows in Novu, orchestrate notifications across channels.',
+        status: workflows && workflows.totalCount > 0 ? 'completed' : 'in-progress',
+      },
+      {
+        id: `connect-${providerType}-provider` as StepIdEnum,
+        title: getProviderTitle(providerType),
+        description: getProviderDescription(providerType),
+        status: integrations?.some((integration) => isActiveIntegration(integration, providerType))
+          ? 'completed'
+          : 'pending',
+      },
+      {
+        id: StepIdEnum.INVITE_TEAM_MEMBER,
+        title: 'Invite a team member',
+        description: 'Collaborate with your team to manage notifications',
+        status: hasInvitedTeamMember ? 'completed' : 'pending',
+      },
+    ],
+    [workflows, hasInvitedTeamMember, providerType, integrations]
+  );
+
+  useEffect(() => {
+    const previousSteps = previousStepsRef.current;
+
+    // Track step completion changes
+    steps.forEach((step) => {
+      const previousStep = previousSteps.find((prev) => prev.id === step.id);
+      if (previousStep?.status !== 'completed' && step.status === 'completed') {
+        telemetry(TelemetryEvent.WELCOME_STEP_COMPLETED, {
+          stepId: step.id,
+          stepTitle: step.title,
+        });
+      }
+    });
```

Comment:
I guess this should be tracked on the particular step pages, not here, as it feels hacky, and we might get doubled events

Reply:
We also track it on each step page, I've just added it just in case. But you are right it's weird, Ill remove it.